src/azul/terra.py

```diff
@@ -662,9 +662,19 @@ def for_registered_user(cls, authentication: OAuth2) -> 'TDRClient':
     def drs_client(self) -> DRSClient:
         return DRSClient(http_client=self._http_client)
 
-    def get_duos(self, source: SourceRef) -> MutableJSON:
+    def get_duos(self, source: SourceRef) -> Optional[MutableJSON]:
         response = self._retrieve_source(source)
-        duos_id = response['duosFirecloudGroup']['duosId']
-        url = self._duos_endpoint('dataset', 'registration', duos_id)
-        response = self._request('GET', url)
-        return self._check_response(url, response)
+        duos_info = response.get('duosFirecloudGroup') or {}
+        duos_id = duos_info.get('duosId')
+        if duos_id is None:
```

Comment:
```diff
Subject: [PATCH] Fix: Clients from azul.deployment.aws use unrefreshable credentials (#5699)
---
Index: src/azul/terra.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/terra.py b/src/azul/terra.py
--- a/src/azul/terra.py	(revision f235482c1e0c0e43be3105c7ae9cfdd6a0923d72)
+++ b/src/azul/terra.py	(date 1700180871444)
@@ -664,9 +664,9 @@
 
     def get_duos(self, source: SourceRef) -> Optional[MutableJSON]:
         response = self._retrieve_source(source)
-        duos_info = response.get('duosFirecloudGroup') or {}
-        duos_id = duos_info.get('duosId')
-        if duos_id is None:
+        try:
+            duos_id = response['duosFirecloudGroup']['duosId']
+        except KeyError:
             log.warning('Source %r does not specify a DUOS ID', str(source.spec))
             return None
         else:
```

Reply:
This doesn't work because for the 1000G snapshot, `response['duosFirecloudGroup']` is present but `None`, so your code raises and `AttributeError` instead of a `KeyError`.